build_test.go

```diff
@@ -67,43 +61,49 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 	})
 
 	it("returns a result that builds correctly", func() {
-		result, err := build(packit.BuildContext{
-			WorkingDir: workingDir,
-			CNBPath:    cnbDir,
-			Stack:      "some-stack",
-			BuildpackInfo: packit.BuildpackInfo{
-				Name:    "Some Buildpack",
-				Version: "some-version",
-			},
-			Plan: packit.BuildpackPlan{
-				Entries: []packit.BuildpackPlanEntry{},
-			},
-			Layers: packit.Layers{Path: layersDir},
+		it.Before(func() {
+			t.Setenv("BP_NODE_PROJECT_PATH", "some-project-dir")
```

Comment:
@mhdawson According to golang docs, `t.Setenv` cannot be used during parallel tests. Could you explain why you chose `t` over `os`?

Reply:
I likely saw that somewhere else in the tests or dig a google search being new to go in earlier PRs. I re-used that approach in this PR.

I did notice that in the tests I modified in this PR there was the use of os.Setenv but that seems to have the disadvantage that it affects other tests and so you have set/unset, versus t.Setenv which I think should just affect the one test. 

t.Setenv seems to be a newer option which was added in go.1.1.7
From: https://beta.pkg.go.dev/testing@master#T.Setenv
```
Setenv calls os.Setenv(key, value) and uses Cleanup to restore the environment variable to its original value after the test.
```

So from reading that I think t.Setenv is just an eaiser way to call os.Setenv without having to call os.Unset afterwards.

I do see this 
```
Because Setenv affects the whole process, it cannot be used in parallel tests or tests with parallel ancestors.
```

but I think that applies to os.Setenv as well.

Summing up I think that t.Setenv is just a way to use os.Setenv without having to make sure you cleanup afterwards and both t.Setenv and os.Setenv will have the same restriction related to parallel testing.

My guess is the right answer would be to update the existing calls to os.Setenv to use t.Setenv so that there is less code, less change of missing the cleanup.  Not sure if that is something that should be combined into the PR or done as a separate one?

Reply:
That's right. The main (only) benefit of `t.SetEnv()` is that you don't have to remember to clean up after yourself. But that benefit alone is worth doing. It costs nothing to use it, and it's more idiomatic in the golang testing community.

As far as the scope of this PR - I would say that if there are a few existing `os.SetEnv()` in the codebase that you don't touch as part of your changes, use `os.SetEnv()` everywhere in your additions. Then we can make a separate PR addressing just `os.SetEnv()` everywhere.

If however, there's just a small number (or none) of pre-existing `os.SetEnv()`, then fix them up as part of this PR.

Make sense?